cmd/snap-recovery/partition/mkfs.go

```diff
@@ -0,0 +1,36 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package partition
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/gadget"
+)
+
+func makeFilesystem(node, label, filesystem, content string) error {
```

Comment:
Given that there are four strings here how about using a distinct type for the filesystem type argument?
From the looks of the function alone I don't know what the content argument is. Perhaps a docstring would help here.

Reply:
Yeah, that makes sense, I will add the docstring in a followup to avoid another full spread run.